resellers_api.yaml

```diff
@@ -1,83 +1,280 @@
+              required:
+              - name
+              - regulator_reference
+              - reseller_location_id
+              - reseller_organisation_id
+              - key
+            example:
+              name: Alexander Fleming House
+              regulator_reference: CQC45678
+              reseller_location_id: '22'
+              reseller_organisation_id: '101'
+              key: zR7rsYew0Q_yLMk2jLgOvAthAf5n5q11TSYrRjO7aNjQ
+      responses:
+        '200':
+          description: Creates a new Location
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  name:
                     type: string
-                  secondary_key:
+                  type_of_organisation:
+                    type: string
+                  timezone:
                     type: string
+                  reseller_organisation_id:
+                    type: string
+                  created_at:
+                    type: string
+                  locations:
+                    type: array
+                    items:
+                      type: object
+                      properties:
+                        name:
+                          type: string
+                        timezone:
+                          type: string
+                        regulator_reference:
+                          type: string
+                        postcode:
+                          nullable: true
+                        reseller_location_id:
+                          type: string
+                        reseller_reference:
+                          nullable: true
+                        created_at:
+                          type: string
+                      required:
+                      - name
+                      - timezone
+                      - regulator_reference
+                      - postcode
+                      - reseller_location_id
+                      - reseller_reference
+                      - created_at
+                  users:
+                    type: integer
+                required:
+                - name
+                - type_of_organisation
+                - timezone
+                - reseller_organisation_id
+                - created_at
+                - locations
+                - users
               example:
-                primary_key: zR7rsYew0Q_yLMk2jLgOvAthAf5n5q11TSYrRjO7aNjQ
-                secondary_key: KTXKEhwIROxMfMwZ0G7a3wPLVJ7H69jb8blUoHQzsOqA
-  "/v1/r/list_locations":
+                name: Carehome
+                type_of_organisation: Social Care
+                timezone: London
+                reseller_organisation_id: '101'
+                created_at: '2024-02-02T15:40:14Z'
+                locations:
+                - name: Alexander Fleming House
+                  timezone: London
+                  regulator_reference: CQC45678
+                  postcode:
```

Comment:
Not sure why these postcodes are coming up as blank now. Do the tests not work anymore?

Reply:
`/requests/resellers_spec.rb` doesn't have seeded postcodes. `/controllers/resellers_controller_spec.rb` does, but I see now that that file isn't the one read for building the yaml?

So we need to make sure seeded data we want to see is in `spec/requests/resellers_spec.rb`

Reply:
Ah I see. I'll update the request specs :+1: